testplan/testing/multitest/result.py

```diff
@@ -99,7 +101,23 @@ def __exit__(self, exc_type, exc_value, tb):
 assertion_state = threading.local()
 
 
-def assertion(func):
+def mark(func: Callable) -> Callable:
+    """
+    Sets the decorated function's filepath and line-range in assertion state.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        filepath = inspect.getfile(func)
+        lines, start = inspect.getsourcelines(func)
```

Comment:
these values can be work out in the decorator once on decoration time, no need to do inspect an all call of the function